### PR TITLE
Updates to Alum Directory

### PR DESCRIPTION
**Pull request by awilli70**
### client/src/App.js
- now imports the alumniDirectory component
- replaced the placeholder with AlumniDirectory - in the Alumni case, passes isAlumniView=true

### client/src/components/AlumniDirectory
- no longer takes entries prop
- instead, takes isAlumniView - this is decides whether we need to show or hide the request button
- global constant searchOptions - holds all fields for search dropdown
- states can no longer be automatically assigned through prop based values
    This has been addressed in two ways:
        - numEntries and totalPages are initialized to 0.
        - ComponentWillMount now calls a function to make an api request, which in resolving the promise sets the state of numEntries, entries and totalPages 
        - it then proceeds to call another function to populate the gradYear and allText states. ComponentWillMount is now asynchronous to allow for this behavior, and will only run once.

new function: async this.search()
- currently returns the number of matching profiles for all queries. Will soon call a helper function to create the needed cards. Pretty cool, right? (Note: this means that profiles will become a state)

When creating profile cards, automatically adds unique graduation years to gradYears
- This enables the population of the dropdown for Graduation Year search 

Added graduation year to profile cards!
requestVisible now manages the visibility of the Make Request button (on, disabled, hidden)
The search row now depends on the current filter state (switches between dropdown and search)

### Routes:
GET: /alumni/all
returns all alumni, similar to /mongoose-util/allAlumni

Screen Shot 2020-05-20 at 9 14 35 PM

Screen Shot 2020-05-20 at 9 15 20 PM

### async Search behavior
Note that a blank filter field is equivalent to all fields
Screen Shot 2020-05-21 at 9 17 37 PM

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -217,7 +218,8 @@ class App extends Component {
                           navItems={alumniNavBarItems()}
                           activeItem={'alumniDirectory'}
                       />
-                      <div> Alumni Directory! </div>
+
+                      <div> <AlumniDirectory isAlumniView={true}/> </div>
```
Comment (reshadbinharun) [inline client/src/App.js:222]: We can remove the wrapping div!

File: client/src/components/AlumniDirectory.js [hunk 2]
```diff
@@ -1,18 +1,72 @@
...
 export default class AlumniDirectory extends Component {
     state = {
         activePage: 1,
-        numEntries: this.props.entries.length,
+        numEntries: 0,
         isLoading: false,
         value: '',
         pageSize: 3,
-        totalPages: Math.ceil(this.props.entries.length/3)
+        totalPages: Math.ceil(0/3),
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:55]: Can be initialized to 0

File: client/src/components/AlumniDirectory.js [hunk 3]
```diff
@@ -1,18 +1,72 @@
...
+        filter: 'all',
+        year: Number
+    }
+
+    componentWillMount() {
+        this.getEntries().then(result => this.setState({
+            entries: result.alumnus,
+            totalPages: Math.ceil(result.alumnus.length/3),
+            numEntries: result.alumnus.length
+        }))
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:66]: Let's add a `.catch` to console any error here

File: client/src/components/AlumniDirectory.js [hunk 4]
```diff
@@ -1,18 +1,72 @@
...
+    }
+    getEntries() {
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:68]: Let's add the `async` modifier.

File: client/src/components/AlumniDirectory.js [hunk 5]
```diff
@@ -1,18 +1,72 @@
...
+        return makeCall(null, '/alumni/all', 'get')
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:69]: and `await` on makeCall

File: server/routes/alumni.js [hunk 6]
```diff
@@ -68,4 +68,14 @@ router.post('/', async (req, res, next) => {
     }
 });
 
+router.get('/all', async (req, res, next) => {
+    try {
+        const dbData = await alumniSchema.find()
+        res.json({'alumnus' : dbData});
```
Comment (reshadbinharun) [inline server/routes/alumni.js:74]: nit: `alumni:`

**Comment by awilli70** [comment]
In progress - still fixing up some stuff

**Comment by awilli70** [comment]
Should now be fully functional (minus search)

File: client/src/components/AlumniDirectory.js [hunk 7]
```diff
@@ -1,79 +1,266 @@
...
+        totalPages: 0,
+        entries:[],
+        gradYears: [],
+        allText: [],
+        results: 0,
+        filter: 'all'
+    }
+
+    async componentWillMount() {
+        let result = await this.getEntries()
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:64]: We can get rid of all awaits I think, apart from this one...
If you inspect `result` returned here, it should not be a PENDING promise.

File: client/src/components/AlumniDirectory.js [hunk 8]
```diff
@@ -1,79 +1,266 @@
...
+        }
+        await gradYears.sort()
+        await this.setState({
+                                gradYears: gradYears,
+                                allText: allText
+                            })
+    }
+
+    async getEntries() {
+        return await makeCall(null, '/alumni/all', 'get').catch(e => console.log(e))
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:99]: ... and this one.

File: client/src/components/AlumniDirectory.js [hunk 9]
```diff
@@ -1,79 +1,266 @@
...
+    }
+
+    async search(value) {
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:102]: Search I believe does not have to be async!

File: client/src/components/AlumniDirectory.js [hunk 10]
```diff
@@ -1,79 +1,266 @@
...
+        numEntries: 0,
         isLoading: false,
         value: '',
         pageSize: 3,
-        totalPages: Math.ceil(this.props.entries.length/3)
+        totalPages: 0,
+        entries:[],
+        gradYears: [],
+        allText: [],
+        results: 0,
```
Comment (reshadbinharun) [inline client/src/components/AlumniDirectory.js:59]: nit: Can we name this numResults?

**Comment by awilli70** [comment]
Closing pull request - changes have been made, but are no longer being reflected in this PR - opening a new one

**Pull request closed without merging**